### DISCUSSION TITLE
Migrate existing splash screen to Android 12

### DISCUSSION THREAD

**Discussion by haoyuant** [xamarin/Xamarin.Forms#14857] [Q&A]
Hello, according to https://developer.android.com/guide/topics/ui/splash-screen/migrate Android 12 introduced a new splash screen system, which provides a default splash screen for all apps and the existing splash screen will be affected by this behaviour. Does anyone know how to do the migration in Xamarin.Android?

**Comment by lee-m** [comment]
The new AndroidX compat library for splash screens referenced in the official docs is still in alpha so the Xamarin AndroidX project won't include bindings for it until it becomes final so as far as "official" Xamarin support goes, we're in a bit of a limbo until we get official bindings.

I've created my own binding for `androidx.core.splashscreen` based on the latest alpha02 release to play around with if you wanted to try it out - https://github.com/lee-m/Xamarin.AndroidX.Core.SplashScreen. I've been meaning to publish as NuGet package for this once I've had chance to verify it works.

As far as migrating my splash screen, I spent a couple of days experimenting using that binding library I created and whilst I was able to get a splash screen working on Android 12 and earlier which looked the same, there are some odd issues still there such as the screen being partially blacked out in some cases when powering the device on/off so abandoned using it for now.

My existing splash screen is based on the Microsoft docs of a separate activity and overriding `windowBackground` in a custom theme and one snag I hit was that my part of my app's icon and window background colour are the same, which meant that the system generated splash screen based on the launch icon looked a bit rubbish so I worked around it by adding the following style in a `values-v31` folder:

```
<?xml version="1.0" encoding="utf-8"?>
<resources>

	<style name="MainTheme.Splash" parent="Theme.AppCompat.Light.NoActionBar">
		<item name="android:windowBackground">@android:color/transparent</item>
	</style>

</resources>
```

Where `MainTheme.Splash` is also present in `values\styles.xml` and has the rest fo the styling for the non-Android 12 splash screen.
